devel/build/command.py

```diff
@@ -137,6 +138,12 @@ def command(shell: DTShell, args, **kwargs):
                                 f"Detected v{project.format.version}. Please, upgrade your project first.")
                 exit(11)
 
+        # Execute pre-build hook
+        if project.format.version >= 4:
+            for shell_command in project.hooks.hooks['pre-build']:
```

Comment:
according to https://github.com/duckietown/lib-dtproject/pull/5, the object `shell_command` is of type `Hook`, so we need three things. Rename `shell_command` -> `hook`.

Use it as
```
_run_cmd(hook.command)
```

Use the value of `hook.required` somehow here.